File: solutions/ruby/allergies/3/allergies.rb
class Allergies
  ALLERGENS = {
    'eggs':           1,
    'peanuts':        2,
    'shellfish':      4,
    'strawberries':   8,
    'tomatoes':      16,
    'chocolate':     32,
    'pollen':        64,
    'cats':         128
  }.freeze

  def initialize(score)
    @score = score
  end

  def allergic_to?(allergen)
    allergen_mask = ALLERGENS.fetch(allergen.to_sym)
    @score & allergen_mask == allergen_mask
  end

  def list
    ALLERGENS.keys
      .map(&:to_s)
      .select(&method(:allergic_to?))
  end
end

File: solutions/csharp/allergies/1/Allergies.cs
using System;
using System.Collections.Generic;
using System.Linq;

public enum Allergen
{
    Eggs         = 1,
    Peanuts      = 2,
    Shellfish    = 4,
    Strawberries = 8,
    Tomatoes     = 16,
    Chocolate    = 32,
    Pollen       = 64,
    Cats         = 128
}

public class Allergies
{
    private Allergen[] _list;

    public Allergies(int mask)
    {
        _list = ((Allergen[]) Enum.GetValues(typeof(Allergen)))
            .Where(a => ((int) a & mask) == (int) a)
            .ToArray();
    }

    public bool IsAllergicTo(Allergen allergen)
        => _list.Contains(allergen);

    public Allergen[] List()
        => _list;
}